File: Code/Python_Code/30.2.1_Alertor/Alertor.py
#!/usr/bin/env python3
#############################################################################
# Filename    : Alertor.py
# Description : Make a sound and light reminder with a buzzer and infrared obstacle avoidance sensor.
# Author      : www.freenove.com
# modification: 2022/4/20
########################################################################
import RPi.GPIO as GPIO
import time
ledPin = 13       # define ledPin
sensorPin = 12    # define buttonPin
buzzerPin = 11    # define buzzerPin

def setup():
    GPIO.setwarnings(False)
    GPIO.setmode(GPIO.BOARD)         # use PHYSICAL GPIO Numbering
    GPIO.setup(ledPin, GPIO.OUT)     # set ledPin to OUTPUT mode
    GPIO.setup(buzzerPin, GPIO.OUT)   # set buzzerPin to OUTPUT mode
    GPIO.setup(sensorPin, GPIO.IN)     # set sensorPin to INPUT mode

def alarm():
    times=3
    while times:
        GPIO.output(buzzerPin,GPIO.HIGH) # turn on buzzer
        GPIO.output(ledPin,GPIO.HIGH)    # turn on led
        time.sleep(0.05)
        GPIO.output(buzzerPin,GPIO.LOW)  # turn off buzzer
        GPIO.output(ledPin,GPIO.LOW)     # turn on led
        time.sleep(0.05)
        times-=1

def sensorEvent(channel): # When sensor is blocked, this function will be executed
    alarm()
    
def loop():
    GPIO.add_event_detect(sensorPin,GPIO.FALLING,callback = sensorEvent,bouncetime=300)
    while True:
        pass
def destroy():
    GPIO.cleanup()             # Release GPIO resource
if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    setup()
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()

File: Code/C_Code/28.2.1_Alertor/Alertor.c
/**********************************************************************
* Filename    : Alertor.c
* Description : Make a sound and light alarm with a buzzer and u type photoelectric sensor. 
* Author      : www.freenove.com
* modification: 2022/4/20
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>
#define ledPin    0  	//define the ledPin
#define sensorPin 1	//define the buttonPin
#define buzzerPin 2  	//define the buzzerPin

void alarm(int times){
    while(times--){
        digitalWrite(buzzerPin, HIGH);
        digitalWrite(ledPin, HIGH);
        delay(50);
        digitalWrite(buzzerPin, LOW); 
        digitalWrite(ledPin, LOW);
        delay(50);
    }
}

void sensorEven(void){
    alarm(3);
}
void  main(void)
{
    printf("Program is starting ... \n");
    wiringPiSetup();	        //Initialize wiringPi.	
    pinMode(ledPin, OUTPUT);  //Set ledPin to output
    pinMode(sensorPin, INPUT);//Set sensorPin to input
    pinMode(buzzerPin, OUTPUT); 
    delay(100);
    wiringPiISR(sensorPin,INT_EDGE_BOTH,&sensorEven);
    digitalWrite(buzzerPin, LOW); 
    while(1){
    }
}
